lib/widgets/mxc_image.dart

```diff
@@ -251,28 +264,47 @@ class _MxcImageState extends State<MxcImage>
   Widget _buildImageWidget() {
     final data = _imageData;
     final needResize = widget.event != null && !widget.noResize;
-    return data == null || data.isEmpty
+    return filePath == null && data == null
         ? placeholder(context)
         : ClipRRect(
             key: Key('${data.hashCode}'),
             borderRadius: widget.rounded
                 ? BorderRadius.circular(12.0)
                 : BorderRadius.zero,
-            child: Image.memory(
-              data,
-              width: widget.width,
-              height: widget.height,
-              cacheWidth: needResize ? widget.width?.toInt() : null,
-              cacheHeight: needResize ? widget.height?.toInt() : null,
-              fit: widget.fit,
-              filterQuality: FilterQuality.medium,
-              errorBuilder: (context, __, ___) {
-                _isCached = false;
-                _imageData = null;
-                WidgetsBinding.instance.addPostFrameCallback(_tryLoad);
-                return placeholder(context);
-              },
-            ),
+            child: filePath != null && filePath!.isNotEmpty
+                ? Image.file(
+                    File(filePath!),
+                    width: widget.width,
+                    height: widget.height,
+                    cacheWidth: needResize ? widget.width?.toInt() : null,
+                    cacheHeight: needResize ? widget.height?.toInt() : null,
+                    fit: widget.fit,
+                    filterQuality: FilterQuality.medium,
+                    errorBuilder: (context, __, ___) {
+                      _isCached = false;
+                      filePath = null;
+                      WidgetsBinding.instance.addPostFrameCallback(_tryLoad);
+                      return placeholder(context);
+                    },
+                  )
+                : data != null
+                    ? Image.memory(
+                        data,
+                        width: widget.width,
+                        height: widget.height,
+                        cacheWidth: needResize ? widget.width?.toInt() : null,
+                        cacheHeight: needResize ? widget.height?.toInt() : null,
+                        fit: widget.fit,
+                        filterQuality: FilterQuality.medium,
+                        errorBuilder: (context, __, ___) {
+                          _isCached = false;
+                          _imageData = null;
+                          WidgetsBinding.instance
+                              .addPostFrameCallback(_tryLoad);
+                          return placeholder(context);
+                        },
+                      )
+                    : const SizedBox.shrink(),
```

Comment:
can you write methods to simplify the comprehension ? nested ternary conditions are hard to read

Reply:
okay. good idea